### DISCUSSION TITLE
Asynchronous BITO AI

### DISCUSSION THREAD

**Discussion by Arturohgm** [gitbito/bitoai#230] [Q&A] [answered]
I have some questions about BITO AI.
Can queries be made asynchronously? How?
Can they be made synchronously? How?
Or can both options be used at the same time in the query? How?

**Comment by bitoandy** [comment, marked as answer]
hi @Arturohgm,

If you are using Bito AI CLI to make requests, you can make multiple requests by writing a script to do so in parallel.
In case you want to use the prior context for a call make sure to use `-c` option where you can provide a context file. When a context file option is used multiple request context (Q and A) is stored in the context file name you provide and is used by Bito CLI to understand prior context while answering the next question. Make sure that no two request in parallel use the same context file. In case two requests need to use the same context file they should ideally be serialized.

In case you need more details or discuss, please send an email to support@bito.ai and we can schedule a call to go over the questions, use cases and provide details on what's possible and how.

**Reply by Arturohgm** [reply]
Just one question.

If the answer you gave me worked, it was exactly what I was looking for.

However, when I try to combine the code, it gives me everything on a single line instead of giving me the formatted code. In this case, it prints the code continuously and prints the line breaks as '/n' instead of as I expected.

#include <iostream>
#include <vector>
For example:

This is what I hope you give me
int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }

    std::vector<int> fib(n + 1);
    fib[0] = 0;
    fib[1] = 1;

    for (int i = 2; i <= n; ++i) {
        fib[i] = fib[i - 1] + fib[i - 2];
    }

    return fib[n];
}

int main() {
    int n;
    std::cout << "Ingresa un número: ";
    std::cin >> n;
    std::cout << "El " << n << "-ésimo número de Fibonacci es: " << fibonacci(n) << std::endl;
    return 0;
}

But the result it gives me is the following:

#include <iostream>\n#include <vector>\n// Función para calcular el n-ésimo número de Fibonacci de manera iterativa\nint fibonacci(int n) { if (n <= 1) { return n; } std::vector<int> fib(n + 1); fib[0] = 0; fib[1] = 1; for (int i = 2; i <= n; ++i) { fib[i] = fib[i - 1] + fib[i - 2]; } return fib[n]; }\nint main() { int n; std::cout << "Ingresa un número: "; std::cin >> n; std::cout << "El " << n << "-ésimo número de Fibonacci es: " << fibonacci(n) << std::endl; return 0; }
